src/blocksigner.cpp

```diff
@@ -45,15 +46,18 @@ bool CBlockSigner::SignBlock()
 
         if(refBlock.IsTPoSBlock())
         {
-            CKeyID merchantKeyID;
-            if(!refContract.merchantAddress.GetKeyID(merchantKeyID))
+            if(!refContract.scriptMerchantAddress.IsPayToPublicKeyHash()) {
```

Comment:
Would you mind elaborating why is this required?

Reply:
Currently we accept merchant addresses only in P2PKH format, so we avoid signing blocks that would get rejected